src/bitcoin/mod.rs

```diff
@@ -388,6 +406,191 @@ impl BitcoinInterface for d::BitcoinD {
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn spent_coins(
+        &self,
+        outpoints: &[(bitcoin::OutPoint, bitcoin::Txid)],
+    ) -> (Vec<SpentCoin>, Vec<bitcoin::OutPoint>) {
+        let ops: Vec<_> = outpoints.iter().map(|(op, _)| op).copied().collect();
+        let wallet_coins = &self.wallet_coins(Some(&ops));
+        let mut spent = Vec::new();
+        let mut expired_spending = Vec::new();
+
+        for (op, spend_txid) in outpoints {
+            if let Some(w_c) = wallet_coins.get(op) {
+                if w_c.spend_txid != Some(*spend_txid) {
+                    expired_spending.push(*op);
+                }
+                if let Some(block) = w_c.spend_block {
+                    spent.push((*op, *spend_txid, block.height, block.time));
+                }
+            }
+        }
+        (spent, expired_spending)
+    }
+
+    fn genesis_block_timestamp(&self) -> u32 {
+        self.client()
+            .genesis_block_timestamp()
+            .expect("Genesis block timestamp must always be there")
+    }
+
+    fn genesis_block(&self) -> BlockChainTip {
+        self.client()
+            .genesis_block()
+            .expect("Genesis block must always be there")
+    }
+
+    fn chain_tip(&self) -> BlockChainTip {
+        // We want the wallet's local chain tip after syncing.
+        self.wallet_tip()
+    }
+
+    fn is_in_chain(&self, tip: &BlockChainTip) -> bool {
+        // Return `false` if no block at same height as `tip`
+        // is in wallet's local chain.
+        self.is_in_wallet_chain(*tip).unwrap_or_default()
+    }
+
+    /// FIXME: make the Bitcoin backend interface higher level. See the comment in the poller next
+    /// to the `sync_wallet()` call.
+    fn common_ancestor(&self, _tip: &BlockChainTip) -> Option<BlockChainTip> {
+        unreachable!("The common ancestor is returned in `sync_wallet()`. If no reorg was detected then, this method will never be called on an Electrum backend.")
+    }
+
+    fn broadcast_tx(&self, tx: &bitcoin::Transaction) -> Result<(), String> {
+        match self.client().broadcast_tx(tx) {
+            Ok(_txid) => Ok(()),
+            Err(e) => Err(e.to_string()),
+        }
+    }
+
+    fn wallet_transaction(
+        &self,
+        txid: &bitcoin::Txid,
+    ) -> Option<(bitcoin::Transaction, Option<Block>)> {
+        self.wallet_transaction(txid)
+    }
+
+    fn mempool_entry(&self, txid: &bitcoin::Txid) -> Option<MempoolEntry> {
+        self.client().mempool_entry(txid).ok()?
+    }
+
+    fn mempool_spenders(&self, outpoints: &[bitcoin::OutPoint]) -> Vec<MempoolEntry> {
+        self.client()
+            .mempool_spenders(outpoints)
+            .unwrap_or_default()
+    }
+
+    fn sync_progress(&self) -> SyncProgress {
+        // FIXME
+        let blocks = self.chain_tip().height as u64;
+        SyncProgress::new(1.0, blocks, blocks)
```

Comment:
Missed this FIXME. It seems OK to always return 100% for now since the API is bitcoind-specific to mean "blocks/headers". But in the future it would be nice to inform the user about the progress of the sync if it takes a few dozens seconds.

Reply:
I removed the `// FIXME` and added a comment.